transcom/mymove
# B 19919 INT - Diversion Tags for Billable Weights and Review Weights

## [B-19199](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-19919)

## Summary

Adds diversion tags in various places related to payment requests. See the document attached to the agility ticket for a visual list of locations the tags are supposed to be added.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test
There's a lot of steps but I don't know of a simpler way to properly set up a pair of diverted shipments.

1. Access the office app as a TIO.
2. Open a move containing an un-diverted HHG, verify there are no Diversion tags in the Billable Weights card or in the Document Viewer Side Panel when you click review weights.
3. Access the customer app and set up a new move with two HHGs setting the requested pickup date to today.
4. Submit the HHGs, approve them as an SC, then as a TOO, no need to do anything special.
5. Log into the Prime Simulator and find your move (we need to add an actual pick up date to be able to request a diversion)
6. On your first HHG click update shipment and add a scheduled and actual pickup date that is today. Fill out any other required fields.
7. On the second HHG you can fill out all four dates and any other required fields, shouldn't matter what you pick.
8. Back as a TOO open the move and request a diversion on your first HHG.
9. Back in the Prime Simulator update the first shipment with delivery dates and check the Diversion checkbox.
10. Copy the ID of the first HHG
11. Still in the Prime Simulator create a shipment, check the diversion checkbox, paste the first shipment's ID in the Diverted From field and fill out the remaining fields.
12. Back as a TOO approve the now unapproved diverted shipments.
13. Back in the Prime Simulator make sure the 3rd HHG has delivery dates filled out.
14. In the Prime Sim create payment requests for all 3 shipments.
15. Log in as a TIO and open the move, you should see a diversion tag in the Billable Weights card.
16. Click "Review weights"
17. You should see a Diversion tag in the side panel header.
18. Click "Review shipment weights" and check all three shipments
19. The same Diversion tag in the side panel's header should remain visible for the two connected diverted shipments but not for the non-diverted shipment.


### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
